subprojects/core/src/integTest/groovy/org/gradle/process/internal/ExecCancellationIntegrationTest.groovy

```diff
@@ -0,0 +1,169 @@
+/*
+ * Copyright 2018 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.gradle.process.internal
+
+import org.gradle.integtests.fixtures.DirectoryBuildCacheFixture
+import org.gradle.integtests.fixtures.daemon.DaemonClientFixture
+import org.gradle.integtests.fixtures.daemon.DaemonIntegrationSpec
+import org.gradle.internal.jvm.Jvm
+import org.gradle.test.fixtures.ConcurrentTestUtil
+import spock.lang.Unroll
+
+class ExecCancellationIntegrationTest extends DaemonIntegrationSpec implements DirectoryBuildCacheFixture {
```

Comment:
❌ Missing test case: A task that calls `exec` or `javaExec` and swallows the error resulting from cancellation, so that the task looks like it was successful. Such a task should also

- rerun on the next build
- not be cached

Reply:
❌ ignoreExitValue=true is not enough I think. I think the exec method will still throw an exception in that case if you have a cancellation. We need a test that completely swallows any exception thrown from the exec method to ensure that the task is still considered cancelled and not up-to-date.